WIP - Compatibility with elasticsearch 2

giovannialbero1992 opened the pull request:
I've modified code to work with elasticsearch 2.3.1 with elastica 3.1.1. 
This breaks BC in order to comply with ruflin/elastica >= 3.1.

It's working for all the uses in our application with no queries' modification (however raises some deprecation notices)

solivier commented on .travis.yml:
```diff
@@ -5,7 +5,6 @@ cache:
 
 php:
-  - 5.3
   - 5.4
```
Maybe we could also drop the support of php5.4 as it's also not supported anymore ?

solivier commented on composer.json:
```diff
@@ -17,8 +17,8 @@
-        "ruflin/elastica": "~2.1",
-        "psr/log": "~1.0"
+        "psr/log": "~1.0",
+        "ruflin/elastica": "3.1.*"
```
Don't forget to change this line: "php": ">=5.3.2"

giovannialbero1992 replied:
Thanks solivier, I've changed that line. I've leaved reference to php 5.4 to maintain comply with elastica.

nyroDev commented:
I tested it quickly as I have to migrate from 1.x to 2.3 quickly.
My project is not very big and don't use many ES feature, but as far as I can see, this PR works correctly for me.

It would be great if it could be integrated in the main project.

And big thanks to @giovannialbero1992 for his work on that!

Elendev commented:
Same here, I've tested it and as far as I can tell, this PR works perfectly.

Thank you @giovannialbero1992 !

ybert commented:
good job! Thank you @giovannialbero1992 !
It would be great to be merged in main project.

giovannialbero1992 commented:
Good news so :) .... I hope that will be other feedbacks to improve the project with elasticsearch 2.*.

tgabi333 commented:
Thanks @giovannialbero1992 !   elastica 3.2.0 just got out, i maybe worth to skip 3.1 and upgrade to 3.2.

alx69440 commented:
Need !!!

giovannialbero1992 commented:
Ok guys, I've updated composer.json with new stable version of ruflin/elastica 3.2 . I've also rebased master into my branch to obtain green tests with symfony 2.3 . For my project seems to work.

hason commented on composer.json:
```diff
@@ -12,13 +12,13 @@
-        "ruflin/elastica": "~2.1",
-        "psr/log": "~1.0"
+        "psr/log": "~1.0",
+        "ruflin/elastica": "3.2.*"
```
Why not `^3.0`?

tgabi333 replied:
Because elastica is not semver yet
On May 20, 2016 06:55, "Martin Hasoň" <notifications@github.com> wrote:

> In composer.json
> <https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1069#discussion_r63991188>
> :
>
> >          "symfony/framework-bundle": "~2.3|~3.0",
> >          "symfony/console": "~2.1|~3.0",
> >          "symfony/form": "~2.1|~3.0",
> >          "symfony/property-access": "~2.3|~3.0",
> > -        "ruflin/elastica": "~2.1",
> > -        "psr/log": "~1.0"
> > +        "psr/log": "~1.0",
> > +        "ruflin/elastica": "3.2.*"
>
> Why not ^3.0?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1069/files/e8f2dad53e04b230386f0bce5b6ec939e06c0d18#r63991188>
>

guillaume-perreal commented:
Works seamlessly with one of my big project there. It would be nice if it could be merged into master for further testing.

joshuataylor commented:
3.2 of elastica is out as well, so composer.json can be updated to that.

David-Crty commented:
Why this PR not accepted ? It would be great.

The-Don-Himself commented:
How can I test/install the new code. Still kind of a novice. Do I just copy replace the contents within the changed files?

Adirelle commented:
@XWB @merk You seems to be maintainer of the bundle. Could you take a look at this promising PR please ?

whisller commented:
Any news about this PR?

giovannialbero1992 commented:
@David-Crty @Adirelle @whisller ... Seems that it's not interest to merge this PR to main project, so in the next days we will start with a new Bundle from FosElastica's code.

The-Don-Himself commented:
@giovannialbero1992 I support you on that.
PHP 5.5 reaches end of life in a few days (10th July), I find it unbelievable for any proper dev to refuse to update code fearing BC breaks for PHP 5.3.
Don"t get me wrong, I totally appreciate and respect their hard work to bring the bundle thus far, and I urge you to atleast mention the origins in your new fork. I just don't want to be left behind and with the lack of communication on this for months, its feels like being held at ransom.

Adirelle commented:
@The-Don-Himself Moreover, BC could be assumed by semver, e.g. starting a 4.x branch with ES 2 support but without PHP5.3.

kayue commented:
ping the @FriendsOfSymfony maintainer

gbeauvoir commented:
@giovannialbero1992 let us know about a new bundle coming to life!
best

giovannialbero1992 commented:
@gbeauvoir the new bundle it's alive :) ... [https://github.com/fazland/FazlandElasticaBundle](url)

gbeauvoir commented:
@giovannialbero1992 when install finishes it says
`Class 'Fazland\ElasticaBundle\FazlandElasticaBundle' not found`
Instead it imported the Class 'Fazland\ElasticaBundle\FOSElasticaBundle'

giovannialbero1992 commented:
@gbeauvoir, have you done `composer require fazland/elastica-bundle:dev-master` ? Because if you do `composer require fazland/elastica-bundle` composer installs last stable version ^3.1 that has not the same namespace.

pedrobola commented:
I was with follow error

> [Attempted to call an undefined method named "hasFacets" of class "Elastica\ResultSet"]

But I changed the file /fazland/elastica-bundle/Paginator/RawPartialResults.php at line 46 for

`if (method_exists($this->resultSet, 'getFacets') && $this->resultSet->hasFacets()) {`

and now it work fine

kayue commented:
@pedrobola Facets doesn't exist in ElasticSearch 2 anymore. So it is an expected exception I would say.

Adirelle commented:
@kayue @pedrobola So this is probably a bug of fazland/FazlandElasticaBundle, please report it [there](https://github.com/fazland/FazlandElasticaBundle/issues/new).

Bukashk0zzz commented:
@giovannialbero1992 thanks for bundle. 

@FriendsOfSymfony @XWB @merk You not maintain this bundle any more? 
Just create new brunch with PHP 5.5 + ElasticSearch 2 support and tag it. Or add someone write access to repo.
That take 10 minutes of your life.

XWB commented:
I'd be happy to remove PHP 5.3 support but it's up to @merk 

BTW why is this PR so much smaller than https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1002 ? Is it incomplete?

jamyouss commented:
ping @FriendsOfSymfony. Can you merge the PR please !!!!!
@giovannialbero1992 thanks for the bundle.

keanolane commented:
ping :)

XWB commented:
@giovannialbero1992 I pushed 3.2 and updated master to 4.0-dev. Can you rebase this PR from master?

giovannialbero1992 commented:
@XWB I've rebased the PR. I hope that from now you will maintain the bundle actively, because this PR is opened since 20th April and the entire community that use this bundle is locked on an old Elasticsearch's version. However we'll go on with our bundle. Thanks for everything done up to here.

XWB commented:
@giovannialbero1992 Wouldn't it be better if you just become a FriendsOfSymfony collaborateur? I don't see the point of having two bundles with the same functionality.

giovannialbero1992 commented:
@XWB it can be possible, but we would take our modifies done on FazlandElasticaBundle, because we use new functionality implemented on it and without them our project doesn't work (ref. https://github.com/fazland/FazlandElasticaBundle/pull/8) .  
If for you is ok take our active contribution on this project there are no problems. Let me know. Thanks

XWB commented:
@giovannialbero1992 New features are always welcome, as long as they are not proprietary. The new repository manager seems fine to me.

I made you and @alekitto collaborators, I suggest you guys port over the functionally from fazland/FazlandElasticaBundle and then we will works towards a 4.0 release.
